### PR TITLE
Fixed injections

### PR DESCRIPTION
**Pull request by HarrisonS-Phys**

### CODE REVIEW

**Comment by maxisi** [comment]
@HarrisonS-Phys please add a basic description of the changes you have made so I can understand your PR

**Comment by HarrisonS-Phys** [comment]
Previously when performing injections, Ringdown(Signal) and from_parameters() expected a list of all the modes in the fit, but these modes were not being saved as kws in get_template(). Now the modes are properly passed.

Also, when initializing Ringdown(Signal), the way in which the modes were being assigned within the class' namespace led to a fatal error, because indexing.construct(modes) (note: not construct_mode_list(), which is not an existing method), returns  HarmonicIndex.construct(*mode): this passes a HarmonicIndex to the constructor of HarmonicIndex, which throws an error. Now, the mode values are just passed in Ringdown(Signal) as the original list from get_template() to avoid this fatal error.

The rest of the changes in this PR, described below, are minor and do not alter the code's functionality. 

In from_parameters(), the get() method was being implemented in the form get(keyname = "modes", value = None). However, get() defaults to value = None, so assigning it is not necessary and the method is (in my opinion) cleaner to read without the redundancy.

Lastly, in the docstring of the HarmonicIndex construct() method, there was a grammatical error. "An" is the indefinite article that precedes nouns which start with a vowel, however here "an" was being used in front of the open compound noun "black hole", which begins with a consonant and should thus be preceded by "a".

**Comment by HarrisonS-Phys** [comment]
Additionally, I've added the ability to use fit.run() with numpyro.validation_enabled() . This gives extremely helpful error messages when things break. It is currently implemented as an optional feature of fit.run(), but defaults to True.

File: ringdown/waveforms/ringdown.py [hunk 1]
```diff
@@ -13,7 +13,7 @@ class Ringdown(Signal):
 
     def __init__(self, *args, modes=None, **kwargs):
         super(Ringdown, self).__init__(*args, **kwargs)
-        self.modes = indexing.construct_mode_list(modes)
+        self.modes = modes
```
Comment (maxisi) [inline ringdown/waveforms/ringdown.py:17]: modifying here to use the correct ModeIndexList api

**Comment by maxisi** [comment]
Thanks. I am merging this with two modifications: `ModeIndexList` is now properly constructed inside `Ringdown` signal; sampler does not validate arguments by default for performance.
!merge

**Pull request merged**